lib/spack/llnl/util/lang.py

```diff
@@ -374,6 +374,22 @@ def __iter__(self):
     return wrapper()
 
 
+def dedupe(sequence):
+    """Yields a stable de-duplication of an hashable sequence
+
+    Args:
+        sequence: hashable sequence to be de-duplicated
+
+    Returns:
+        stable de-duplication of the sequence
+    """
+    seen = set()
+    for x in sequence:
+        if x not in seen:
+            yield x
+            seen.add(x)
```

Comment:
I don't really understand this. How is this different from a regular set? Why not just call `set()` above instead of `list(dedupe())`?